## Statistical Analysis: Hypothesis Testing on Poverty

In [1]:
import pandas as pd
from scipy.stats import pearsonr, spearmanr

# Load dataset
df = pd.read_csv("../../data/clean/log_master_dataset_2015_2020.csv")

# Target variable
target_col = "poverty_percent"

# Feature groups
feature_groups = {
    "Socially vulnerable": [
        "children_percent", "elderly_percent", "disabled_rate_per_1000",
        "disabled_60_plus_rate", "disabled_51_60_rate", "disabled_41_50_rate",
        "disabled_31_40_rate", "disabled_18_30_rate"
    ],
    "Economy (consumption/production)": [
        "production_per_capita", "retail_per_capita"
    ],
    "Additional indicators": [
        "alcohol_rate", "drugs_rate", "addiction_rate",
        "infant_mortality_urban_rate", "infant_mortality_rural_rate"
    ]
}

# Function to compute correlations
def run_correlation_tests(df, target_col, feature_groups):
    results = []
    for group_name, features in feature_groups.items():
        for feature in features:
            pearson_corr, pearson_p = pearsonr(df[feature], df[target_col])
            spearman_corr, spearman_p = spearmanr(df[feature], df[target_col])
            results.append({
                "Group": group_name,
                "Feature": feature,
                "Pearson r": round(pearson_corr, 3),
                "p-value (Pearson)": round(pearson_p, 4),
                "Spearman ρ": round(spearman_corr, 3),
                "p-value (Spearman)": round(spearman_p, 4)
            })
    return pd.DataFrame(results)

# Run analysis
results_df = run_correlation_tests(df, target_col, feature_groups)
print(results_df)

                               Group                      Feature  Pearson r  \
0                Socially vulnerable             children_percent      0.028   
1                Socially vulnerable              elderly_percent      0.055   
2                Socially vulnerable       disabled_rate_per_1000      0.143   
3                Socially vulnerable        disabled_60_plus_rate      0.038   
4                Socially vulnerable          disabled_51_60_rate      0.238   
5                Socially vulnerable          disabled_41_50_rate      0.305   
6                Socially vulnerable          disabled_31_40_rate      0.268   
7                Socially vulnerable          disabled_18_30_rate      0.302   
8   Economy (consumption/production)        production_per_capita     -0.319   
9   Economy (consumption/production)            retail_per_capita     -0.735   
10             Additional indicators                 alcohol_rate     -0.092   
11             Additional indicators    

Within the framework of this thesis project, a statistical analysis was conducted to evaluate the relationships between poverty levels and socio-economic features.

---

### 1. Do children, elderly, and disabled populations affect poverty?

Correlation tests (Pearson and Spearman) were performed between the share of socially vulnerable groups and poverty level (`poverty_percent`):

- Disabled (ages 51–60): r = 0.238, p < 0.0001  
- Overall disability rate: r = 0.143, p = 0.0012  

Conclusion: the share of people with disabilities has a statistically significant effect on poverty levels. Other groups (children, elderly) did not show significant relationships.

---

### 2. Is poverty related to production and consumption?

Correlations were computed between poverty and economic indicators:

- `retail_per_capita`: r = –0.735, p < 0.0001  
- `production_per_capita`: r = –0.319, p < 0.0001  

Conclusion: higher levels of consumption and production are associated with lower poverty. This highlights the importance of economic activity for regional well-being.

---

### 3. What other relationships are observed?

Analysis of additional indicators showed:

- Rural infant mortality: r = 0.476, p < 0.0001  
- Alcohol, drug use, and overall addiction — not statistically significant  

Conclusion: poverty is strongly associated with higher rural infant mortality, indicating worsening living conditions among the most vulnerable populations.

---

### Conclusion

The conducted tests confirmed several significant relationships that help explain the drivers of poverty across Russian regions.  
These findings will be useful for selecting key features for clustering and for interpreting the results.